src/lab1/TestMovie.java

```diff
@@ -0,0 +1,160 @@
+                    movieIndex = searchMovie(allMovies, movieName);
+                    System.out.println(allMovies[movieIndex].getRating());
+                    break;
+                case "6":
+                    System.out.print("Enter the movie you want to update: ");
+                    movieName = SC.nextLine();
+                    movieIndex = searchMovie(allMovies, movieName);
+                    if (movieIndex != -1) {
+                        System.out.print("Update the rating: ");
+                        allMovies[movieIndex].setRating(Double.parseDouble(SC.nextLine()));
+                    }
+                    break;
+                case "7":
+                    System.out.print("Enter the name of the movie that you want to delete: ");
+                    movieName = SC.nextLine();
+                    movieIndex = searchMovie(allMovies, movieName);
+                    if (movieIndex != -1) {
+                        allMovies = removeMovie(allMovies, movieIndex);
+                    }
+                    break;
+                case "8":
+                    sort(allMovies);
+                    break;
+                case "0":
+                    appIsOn = false;
+                    break;
+            }
+        }
+    }
+
+    public static void sort(Movie[] allMovies) {
+        for (int i = 0; i < allMovies.length - 1; i++) {
+            for (int j = 0; j < allMovies.length - 1 - i; j++) {
+                if (allMovies[j].getRating() > allMovies[j + 1].getRating()) {
+                    Movie temp = allMovies[j];
+                    allMovies[j] = allMovies[j + 1];
+                    allMovies[j + 1] = temp;
+                }
+            }
+        }
+    }
+
+    public static int searchMovie(Movie[] allMovies, String movie) {
+        for (int i = 0; i < allMovies.length; i++) {
+            if (allMovies[i].getName().equals(movie)) {
+                return i;
+            }
+        }
+        System.out.println("Movie not found");
+        return -1;
+    }
+
+    public static String[] getMovieAndRatingFromUser() {
+        System.out.print("Enter the movie: ");
+        String movieName = SC.nextLine();
+        System.out.print("Enter the rating of the movie: ");
+        String movieRating = SC.nextLine();
+        return new String[]{movieName, movieRating};
+    }
+
+    public static Movie[] mergeArrays(Movie[] baseArray, Movie[] newArray) {
+        int size = baseArray.length + newArray.length;
+        Movie[] mergedArray = new Movie[size];
+        System.arraycopy(baseArray, 0, mergedArray, 0, baseArray.length);
+        System.arraycopy(newArray, 0, mergedArray, baseArray.length, newArray.length);
+        return mergedArray;
+    }
+
+    public static void printMovies(Movie[] allMovies) {
+        for (Movie movie : allMovies) {
+            System.out.println(movie);
+        }
+    }
+
+    public static Movie[] removeMovie(Movie[] allMovies, int index) {
```

Comment:
try to use System.arraycopy() because it is `@IntrinsicCandidate` if don't know what it is read about it)